crates/ra_ide_api/src/db.rs

```diff
@@ -90,3 +106,16 @@ fn line_index(db: &impl ra_db::SourceDatabase, file_id: FileId) -> Arc<LineIndex
     let text = db.file_text(file_id);
     Arc::new(LineIndex::new(&*text))
 }
+
+#[derive(Debug, Default, Clone)]
+pub(crate) struct DebugData {
+    pub(crate) root_paths: FxHashMap<SourceRootId, String>,
+    pub(crate) crate_names: FxHashMap<CrateId, String>,
```

Comment:
wouldn't we want to have access to these things in some non-debugging cases as well?

Reply:
I think only for debuginfo generation (which is far off) and for error reporting (which happens outside of hir). So yeah, I think we should maybe name this thing differently, but it shouldn't be accessible to HIR in the same way the source database is accessible .